src/application.ts

```diff
@@ -195,26 +210,35 @@ export class Application {
     // if installation ID passed, instantiate and authenticate Octokit, then cache the instance
     // so that it can be used across received webhook events.
     if (id) {
-      // Cache for 1 minute less than GitHub expiry
-      const installationTokenTTL = parseInt(process.env.INSTALLATION_TOKEN_TTL || '3540', 10)
+      const options = {
+        auth: async () => {
+          const accessToken = await this.app.getInstallationAccessToken({ installationId: id })
+          return `token ${accessToken}`
+        },
+        baseUrl: process.env.GHE_HOST && `https://${process.env.GHE_HOST}/api/v3`,
+        logger: log.child({ name: 'github', installation: String(id) })
+      }
 
-      return this.cache.wrap(`app:${id}`, async () => {
+      if (this.throttleOptions) {
         return GitHubAPI({
-          auth: async () => {
-            const accessToken = await this.app.getInstallationAccessToken({ installationId: id })
-            return `token ${accessToken}`
-          },
-          baseUrl: process.env.GHE_HOST && `https://${process.env.GHE_HOST}/api/v3`,
-          logger: log.child({ name: 'github', installation: String(id) })
+          ...options,
+          throttle: {
+            id,
+            ...this.throttleOptions
+          }
         })
-      }, { ttl: installationTokenTTL })
+      }
+
+      // Cache for 1 minute less than GitHub expiry
+      const installationTokenTTL = parseInt(process.env.INSTALLATION_TOKEN_TTL || '3540', 10)
+      return this.cache.wrap(`app:${id}`, () => GitHubAPI(options), { ttl: installationTokenTTL })
     }
 
     const token = this.githubToken || this.app.getSignedJsonWebToken()
     const github = GitHubAPI({
       auth: `Bearer ${token}`,
       baseUrl: process.env.GHE_HOST && `https://${process.env.GHE_HOST}/api/v3`,
-      logger: log.child({ name: 'github', installation: String(id) })
+      logger: log.child({ name: 'github' })
```

Comment:
that change should have happened with the last refactoring. `id` is not set as we check for it above, so we can as well just remove it